### PR TITLE
Add environment variable support

### LINKED ISSUE #13: feature: Add vim path variable  completion support
Thanks for your work! 
Can you add vim variable path completion support?
For example, here's a path variable named `$VIM_PATH`, which is `let $VIM_PATH = fnamemodify(resolve(expand('<sfile>:p')), ':h:h')
`, I want to get completion infomation when I type `VIM_PATH/`, any idea for this? Thanks.

### PR DESCRIPTION
**Pull request by Shougo**
Fix #13 

I have added environment variable support.

Note: It uses Deno env instead of Vim env.  Which is better?

### CODE REVIEW

**Comment by Shougo** [comment]
I have pushed Vim's environment version.

**Comment by LumaKernel** [comment]
I'm going to merge vim's env version.
I pushed 1 commit to support 1) windows sep (it should be escaped) 2) windows style env (powershell and cmd).

File: autoload/ddc_file/internal.vim [hunk 1]
```diff
@@ -2,8 +2,8 @@ function! s:line_to_file_full(line, is_posix) abort
   return matchlist(
     \ a:line,
     \ a:is_posix
-       \ ? '[[:fname:]/]*$'
-       \ : '[[:fname:]\\]*$'
+       \ ? '\c\%(^\$\%(env:\)\?\)\?[[:fname:]/]*$'
+       \ : '\c\%(^\$\%(env:\)\?\)\?[[:fname:]\\]*$'
```
Comment (LumaKernel) [inline autoload/ddc_file/internal.vim:6]: You may think this should be done by vim users themselves, but if we edit .ps1 file in linux, fname does not include colon by default, and detecting colon **only in head** does not affect so destructively.

**Comment by LumaKernel** [comment]
image

😄 

Can you give me a quick review?

**Comment by Shougo** [comment]
It looks good.  Thank you!

**Pull request merged**